tracks/swift/exercises/difference-of-squares/mentoring.md

```diff
@@ -0,0 +1,54 @@
+# Difference of Squares
+
+_Differences of Squares_ is a core exercise, unlocked after _Gigasecond_.
+
+## Reasonable Solutions
+
+```swift
+class Squares {
+    let maxNum: Int
+    
+    lazy var squareOfSum: Int = {
+        let sum = (0...self.maxNum).reduce(0, { $0 + $1 })
+        return sum.squared
+    }()
+    
+    lazy var sumOfSquares: Int = {
+        return (0...self.maxNum).reduce(0, { $0 + $1.squared })
+    }()
+    
+    lazy var differenceOfSquares: Int = {
+        return self.squareOfSum - self.sumOfSquares
+    }()
+    
+    init(_ maxNum: Int) {
+        self.maxNum = maxNum
+    }
+}
+
+extension Numeric {
+    var squared: Self {
+        return self * self
+    }
+}
+```
+
+## Common suggestions
+
+- Sometimes students will try to do all of the computations within the init method. This is not optimal when numbers get very large; the cost of computation starts to rise. So the use of lazy variables is strongly encouraged so that computation is only done if needed.
+- Encourage the use of reduce here. It's a perfect place to understand exactly what this function does and how powerful it can be.
+- In addition it is a great place to teach students about the power of ranges. You can assemble a range in a variety of ways, however the syntactial sugar of `...` is the simpliest and most straight forward.
+
+## Next Level Additions
```

Comment:
Have you considered the format of mentor notes described [here](https://github.com/exercism/website-copy/blob/master/README.md#mentor-notes)?

Reasonable solutions, common suggestions, talking points.

This looks like it could be another reasonable solution, since it's so complete. But you could have advanced alternatives as a talking point, too.

Reply:
Thanks for pointing me there, I think I cleaned it up now to the correct structure.